auth/basic/db.go

```diff
@@ -115,3 +141,39 @@ func (b *Store) Validate(ctx context.Context, username, password string) (string
 
 	return userID, nil
 }
+
+// ValidatePassword should check if the password matches the user's stored password
+func (b *Store) ValidatePassword(ctx context.Context, userID string, password string) (ValidatedPassword, error) {
+	err := validate.Many(
+		validate.UUID("UserID", userID),
+		validate.Text("oldPassword", password, 8, 200),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	row := b.getByID.QueryRowContext(ctx, userID)
+	var hash string
+	err = row.Scan(&hash)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, errors.New("invalid userID")
+		}
+		return nil, errors.WithMessage(err, "user lookup failure")
+	}
+
+	b.mx.Lock()
+	defer b.mx.Unlock()
+
+	err = bcrypt.CompareHashAndPassword([]byte(hash), []byte(password))
+	if err != nil {
+		return nil, validation.NewFieldError("oldPassword", "invalid password")
+	}
+
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(password), 12)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
We don't need this for validation, and it results in slowing things down by double.

We only do that in the new password because we need to store the generated hash. For verification, we just want to ensure we validated for the correct user.